### ISSUE TITLE
NullPointerException thrown from ProfileEventAdapter.completed() under certain circumstances

### ISSUE THREAD

**Issue by eriwen** [gradle/gradle#1247]
It seems that an NPE is thrown if file permissions prevent the gradle build from accessing or deleting outputs under `buildSrc/build` (and probably other scenarios).

### Context
[GRADLE-3200](https://issues.gradle.org/browse/GRADLE-3200) from JIRA. 

Stacktrace in brief:
```
java.lang.NullPointerException
at org.gradle.profile.ProfileEventAdapter.completed(ProfileEventAdapter.java:72)
at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
at org.gradle.listener.BroadcastDispatch.dispatch(BroadcastDispatch.java:83)
at org.gradle.listener.BroadcastDispatch.dispatch(BroadcastDispatch.java:31)
```

### Steps to Reproduce (for bugs)
**This is necessary to be actionable and has not been produced yet**

### Your Environment
Gradle 2.0, 2.1

**Comment by linkinpark342** [comment]
This appears to be resolved somewhere between gradle 2.3 and 3.3. The output is not the significantly more reasonable:

```
$ ./gradlew
Using gradle at '/tmp/test/gradlew' to run buildfile '/tmp/test/build.gradle':


FAILURE: Build failed with an exception.

* What went wrong:
java.io.FileNotFoundException: /private/tmp/test/buildSrc/.gradle/noVersion/buildSrc/cache.properties.lock (Permission denied)
> /private/tmp/test/buildSrc/.gradle/noVersion/buildSrc/cache.properties.lock (Permission denied)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1.101 secs
```

This is much clearer than an NPE. Thank you very much, whoever ended up fixing this :)

**Comment by eriwen** [comment]
@linkinpark342 Very glad to hear. I'll close this then — thanks for the quick response!
